rosidl_generator_c/cmake/rosidl_generator_c_generate_interfaces.cmake

```diff
@@ -174,29 +174,46 @@ if(NOT rosidl_generate_interfaces_SKIP_INSTALL)
   ament_export_include_directories(include)
 endif()
 
-if(NOT "${_generated_msg_headers}${_generated_msg_sources}${_generated_srv_headers}${_generated_srv_sources} " STREQUAL " ")
-  find_package(ament_cmake_cppcheck)
-  if(ament_cmake_cppcheck_FOUND)
-    ament_cppcheck(
-      TESTNAME "cppcheck_rosidl_generated_c"
-      "${_output_path}")
-  endif()
-  find_package(ament_cmake_cpplint)
-  if(ament_cmake_cpplint_FOUND)
-    get_filename_component(_cpplint_root "${_output_path}" DIRECTORY)
-    ament_cpplint(
-      TESTNAME "cpplint_rosidl_generated_c"
-      # the generated code might contain longer lines for templated types
-      MAX_LINE_LENGTH 999
-      ROOT "${_cpplint_root}"
-      "${_output_path}")
-  endif()
-  find_package(ament_cmake_uncrustify)
-  if(ament_cmake_uncrustify_FOUND)
-    ament_uncrustify(
-      TESTNAME "uncrustify_rosidl_generated_c"
-      # the generated code might contain longer lines for templated types
-      MAX_LINE_LENGTH 999
-      "${_output_path}")
+if(BUILD_TESTING)
+  if(NOT "${_generated_msg_headers}${_generated_msg_sources}${_generated_srv_headers}${_generated_srv_sources} " STREQUAL " ")
+    find_package(ament_cmake_cppcheck QUIET)
+    if(ament_cmake_cppcheck_FOUND)
+      ament_cppcheck(
+        TESTNAME "cppcheck_rosidl_generated_c"
+        "${_output_path}")
+    else()
+      message(WARNING
```

Comment:
I agree, option b doesn't make sense. Either it's a core package and we want to ensure the linting is being done, or it's a user's message package and they probably don't care about linting their generated message code. That's because there is good coverage on the generated code in the core message packages and it makes running all your tests slightly longer.